pkg/agents/team2/qtable.go

```diff
@@ -0,0 +1,42 @@
+package team2
+
+type maxQAction struct {
+	bestAction int
+	maxQ       float32
+}
+
+func max(a float32, b float32) float32 {
+	if a >= b {
+		return a
+	} else {
+		return b
+	}
+}
```

Comment:
If you use `float64` (which you probably should) then you can use `math.Max()`

Reply:
Used float32 to reduce memory consumption, as the tables will expand exponentially in future versions.

Reply:
I agree reducing the type can reducing memory consumption, just how big is the table? a few bytes or way more?

Reply:
Depends on how well we want the agent to learn. The current plan is to limit it to 1000 elements per table. If that is not enough we might opt for neural networks that can replace tabular methods.

Reply:
I think using either wouldn't matter because it's a x2. If the scaling up is exponential, there will be a memory problem regardless of the type after a certain scale degree.

That being said, keeping `float32` isn't really an issue imo